src/actions/mintArveaweNFT.ts

```diff
@@ -0,0 +1,266 @@
+import { calculate } from '@metaplex/arweave-cost';
+import { LAMPORTS_PER_SOL, PublicKey } from '@solana/web3.js';
+import crypto from 'crypto';
+import { Connection } from '../Connection';
+import { MetadataJson } from '../types';
+import { Wallet } from '../wallet';
+import { ArweaveStorage, ArweaveUploadResult } from '../providers';
+import { PayForFiles } from '../programs';
+import { prepareTokenAccountAndMintTx } from './shared';
+import { Metadata, Creator, MetadataDataData, CreateMetadata } from '../programs/metadata';
+import { sendTransaction } from './transactions';
+import { updateMetadata } from './updateMetadata';
+import { createMasterEdition } from './createMasterEdition';
+
+export enum ENftProgress {
+  none,
+  minting,
+  preparing_assets,
+  signing_metadata_transaction,
+  sending_transaction_to_solana,
+  waiting_for_initial_confirmation,
+  waiting_for_final_confirmation,
+  uploading_to_arweave,
+  updating_metadata,
+  signing_token_transaction,
+}
+
+export async function convertFile2Hash(file: File): Promise<Buffer> {
+  const hashSum = crypto.createHash('sha256');
+  const data = await file.text();
+  hashSum.update(data);
+  const hex = hashSum.digest('hex');
+  return Buffer.from(hex);
+}
+
+export function getAssetCostInfo(files: Map<string, Buffer>) {
+  const sizes = Array.from(files.values()).map((f) => f.byteLength);
+  return calculate(sizes);
+}
+
+export function createFilePack(metadata: MetadataJson, WebFile = File): File {
+  const filedata = {
+    animation_url: undefined,
+    attributes: metadata.attributes && metadata.attributes.length ? metadata.attributes : undefined,
+
+    name: metadata.name,
+    symbol: metadata.symbol,
+    description: metadata.description,
+    seller_fee_basis_points: metadata.seller_fee_basis_points,
+
+    image: metadata.image,
+    external_url: metadata.external_url,
+    properties: {
+      files: metadata.properties.files,
+      category: metadata.properties.category,
+      creators: metadata.properties.creators.map(({ address, share }) => ({
+        address,
+        share,
+      })),
+    },
+  };
+
+  return new WebFile([JSON.stringify(filedata)], 'metadata.json');
+}
+
+export async function getCostInfo(
+  file: File,
+  metadata: MetadataJson,
+  fileMetadata = createFilePack(metadata),
+) {
+  const [dataFile, dataMetadata] = await Promise.all([
+    file.arrayBuffer().then((d) => Buffer.from(d)),
+    fileMetadata.arrayBuffer().then((d) => Buffer.from(d)),
+  ]);
+  const files = new Map<string, Buffer>();
+  files.set(file.name, dataFile);
+  files.set(fileMetadata.name, dataMetadata);
+  return getAssetCostInfo(files);
+}
+export async function getCost(
+  file: File,
+  metadata: MetadataJson,
+  fileMetadata = createFilePack(metadata),
+) {
+  const calcInfo = await getCostInfo(file, metadata, fileMetadata);
+  return calcInfo.solana * LAMPORTS_PER_SOL;
+}
+
+const ARWEAVE_WALLET = new PublicKey('6FKvsq4ydWFci6nGq9ckbjYMtnmaqAoatz5c9XWjiDuS');
```

Comment:
I think it's worth updating const in https://github.com/metaplex/js/blob/main/src/config.ts and utilizing it here.